vllm/transformers_utils/configs/speculators_eagle.py

```diff
@@ -0,0 +1,284 @@
+    def _convert_speculators_to_vllm(cls, speculators_config: dict[str, Any]) -> dict[str, Any]:
+        """
+        Convert speculators Eagle config format to vLLM format.
+        
+        This method handles the translation of field names and structure
+        between speculators and vLLM formats. It supports both Eagle-1
+        and Eagle-3 variants based on speculators_model_type.
+        
+        Args:
+            speculators_config: Dictionary containing speculators format config
+        
+        Returns:
+            Dictionary with vLLM-compatible Eagle configuration
+        """
+        speculators_model_type = speculators_config["speculators_model_type"]
+        transformer_config = speculators_config["transformer_layer_config"]
+        
+        # Extract num_lookahead_tokens from proposal_methods
+        num_lookahead_tokens = cls._extract_num_lookahead_tokens(speculators_config)
+        
+        # Build base vLLM config
+        vllm_config = {
+            "model": transformer_config,
+            "method": speculators_model_type,  # Use speculators_model_type as method
+            "num_lookahead_tokens": num_lookahead_tokens,
+        }
+        
+        # Apply version-specific conversions
+        if speculators_model_type == "eagle":
+            cls._apply_eagle_v1_config(speculators_config, transformer_config, vllm_config)
+        elif speculators_model_type == "eagle3":
+            cls._apply_eagle_v3_config(speculators_config, transformer_config, vllm_config)
+        
+        # Ensure transformer config has required fields
+        cls._ensure_transformer_architectures(speculators_config, transformer_config)
+        
+        # Preserve additional fields not handled by specific conversions
+        cls._preserve_additional_fields(speculators_config, vllm_config)
+        
+        return vllm_config
+    
+    @classmethod
+    def _extract_num_lookahead_tokens(cls, config: dict[str, Any]) -> int:
+        """
+        Extract number of lookahead tokens from proposal methods.
+        
+        Args:
+            config: Speculators config dictionary
+        
+        Returns:
+            Number of speculative tokens
+        
+        Note:
+            Currently only supports the first proposal method.
+            Future versions may support multiple proposal methods.
+        """
+        speculators_cfg = config["speculators_config"]
+        proposal_methods = speculators_cfg["proposal_methods"]
+        
+        # Currently we only support one proposal method
+        first_method = proposal_methods[0]
+        num_lookahead_tokens = first_method.get("speculative_tokens")
+        
+        if num_lookahead_tokens is None:
+            raise ValueError(
+                "Missing 'speculative_tokens' in proposal method. "
+                f"Got: {first_method}"
+            )
+        
+        return num_lookahead_tokens
+    
+    @classmethod
+    def _apply_eagle_v1_config(
```

Comment:
these are fine for now but wondering if we can combine the eagle1/eagle3 _apply methods defined here